clients/StateChannelWallet.ts

```diff
@@ -276,18 +276,18 @@ export class StateChannelWallet {
   }
 
   // Craft an HTLC struct, put it inside a state, hash the state, sign and return it
-  addHTLC(amount: bigint, hash: string): SignedState {
+  async addHTLC(amount: bigint, hash: string): Promise<SignedState> {
     const currentTimestamp: number = Math.floor(Date.now() / 1000); // Unix timestamp in seconds
 
     if (this.myRole() === Participant.Intermediary) {
-      if (Number(this.currentState().intermediaryBalance) < BigInt(amount)) {
+      if (BigInt(this.currentState().intermediaryBalance) < BigInt(amount)) {
         throw new Error("Insufficient balance");
       }
     }
 
     if (
       this.myRole() === Participant.Owner &&
-      Number(this.getOwnerBalance()) < BigInt(amount)
+      (await this.getOwnerBalance()) < amount
```

Comment:
Here we were actually casting a promise to a number, which results in `nan < amount` which is always false. So this check was completely broken.

Reply:
Yowsa. Good catch.

Because we're good wallet developers, I think there's at least a 60% chance that the intermediary would reject HTLCs that should have failed here.